packages/libs/lambda-at-edge/src/default-handler.ts

```diff
@@ -602,12 +602,95 @@ const handleOriginResponse = async ({
   const request = event.Records[0].cf.request;
   const { uri } = request;
   const { status } = response;
+  const { region, domainName } = request.origin?.s3 || {};
+  const bucketName = domainName?.replace(`.s3.${region}.amazonaws.com`, "");
+
   if (status !== "403") {
     // Set 404 status code for 404.html page. We do not need normalised URI as it will always be "/404.html"
     if (uri.endsWith("/404.html")) {
       response.status = "404";
       response.statusDescription = "Not Found";
+      return response;
+    }
+
+    const initialRevalidateSeconds =
+      manifest.pages.ssg.nonDynamic?.[uri.replace(".html", "")]
+        ?.initialRevalidateSeconds;
+    const lastModifiedHeaderString =
+      response.headers?.["last-modified"]?.[0]?.value;
+    const lastModifiedAt = lastModifiedHeaderString
+      ? new Date(lastModifiedHeaderString)
+      : null;
+    if (typeof initialRevalidateSeconds === "number" && lastModifiedAt) {
+      const createdAgo =
+        // LastModified should always be defined
+        (Date.now() - (lastModifiedAt.getTime() || Date.now())) / 1000;
+
+      const timeToRevalidate = Math.floor(
+        initialRevalidateSeconds - createdAgo
+      );
+
+      response.headers["cache-control"] = [
+        {
+          key: "Cache-Control",
+          value:
+            timeToRevalidate < 0
+              ? "public, max-age=0, s-maxage=0, must-revalidate"
+              : `public, max-age=0, s-maxage=${timeToRevalidate}, must-revalidate`
+        }
+      ];
+
+      if (timeToRevalidate < 0) {
+        const { SQSClient, SendMessageCommand } = await import(
```

Comment:
As per here: https://github.com/serverless-nextjs/serverless-next.js/pull/1021 actually I don't think Rollup is bundling dynamic imports efficiently (might be a problem with Rollup or maybe we are not using it correctly)? Could you check and maybe adopt the same strategy as the latest revision of this PR and see if the generated code is a true dynamic import

Reply:
Cool - I'll have a play.

Reply:
Assuming this is resolved as per #1029, also seems to have proper split bundles and dynamic imports after pulling in the changes in master.